src/Nethermind/Nethermind.Blockchain/Processing/BlockProcessor.cs

```diff
@@ -212,6 +213,7 @@ private TxReceipt[] ProcessTransactions(Block block, ProcessingOptions processin
 
         private (Block Block, TxReceipt[] Receipts) ProcessOne(Block suggestedBlock, ProcessingOptions options, IBlockTracer blockTracer)
         {
+            _witnessCollector.Reset();
```

Comment:
I wonder if its better to add things to BlockProcessor or maybe add it in independent component and use events like (Start)BlockProcessing and BlockProcessed. Make it more plugin-like.